src/tests/routers/resource_routers/test_router_organisation.py

```diff
@@ -97,3 +101,214 @@ def test_ai_resource_contacts_field_is_ignored(
+    organisation: Organisation
+    ):
+
+    identifier = register_asset(organisation)
+
+    large_content = b"x" * (1 * 1024 * 1024 + 1)
+    large_image = io.BytesIO(large_content)
+    large_image.name = "big_logo.png"
+
+    with logged_in_user():
+        response = client.post(
+            f"/organisations/{identifier}/image",
+            params={"name": "big_logo"},
+            files={"file": ("big_logo.png", large_image, "image/png")},
+            headers={"Authorization": "Fake token"},
+        )
+
+    assert response.status_code == 413
+    assert response.json()["detail"] == "File too large (max 1MB)."
+
+
+def test_organisation_post_image_incorrect_type(client: TestClient, organisation: Organisation):
+
+    identifier = register_asset(organisation)
+    pdf_data = io.BytesIO(b"%PDF-1.4 test-pdf content")
+
+    with logged_in_user():
+        response = client.post(
+            f"/organisations/{identifier}/image",
+            params={"name": "wrong_logo_type"},
+            files={"file": ("wrong_logo_type.pdf", pdf_data, "application/pdf")},
+            headers={"Authorization": "Fake token"},
+        )
+
+    assert response.status_code == 415
+    assert response.json()["detail"] == f"Unsupported file type application/pdf. Allowed image types: {ALLOWED_IMAGE_TYPES}."
+
+
+def test_organisation_put_image(
+    client: TestClient,
+    organisation: Organisation,
+    ):
+
+    identifier = register_asset(organisation)
+
+    fake_image = io.BytesIO(b"\x89PNG\r\n\x1a\n...")  # fake PNG bytes
+    fake_image.name = "logo.png"
+
+    with logged_in_user():
+        response = client.post(
+            f"/organisations/{identifier}/image",
+            params={"name": "logo"},
+            files={"file": ("logo.png", fake_image, "image/png")},
+            headers={"Authorization": "Fake token"},
+        )
+
+        response = client.put(
+            f"/organisations/{identifier}/image",
+            params={"name": "logo"},
+            files={"file": ("logo.png", fake_image, "image/png")},
+            headers={"Authorization": "Fake token"},
+        )
+
+        assert response.status_code == 200, response.json()
+
+def test_organisation_put_image_non_existent(
+    client: TestClient,
+    organisation: Organisation,
+    ):
+
+    identifier = register_asset(organisation)
+
+    fake_image = io.BytesIO(b"\x89PNG\r\n\x1a\n...")  # fake PNG bytes
+    fake_image.name = "logo.png"
+
+    with logged_in_user():
+
+        response = client.put(
+            f"/organisations/{identifier}/image",
+            params={"name": "LOGO"},
+            files={"file": ("logo.png", fake_image, "image/png")},
+            headers={"Authorization": "Fake token"},
+        )
+        assert response.status_code == 404
+        assert response.json()["detail"] == "No image with the name 'LOGO' found in the database."
+
+
+def test_organisation_get_with_and_without_image(client: TestClient, organisation: Organisation):
```

Comment:
```suggestion
def test_organisation_get_with_and_without_image(client: TestClient, organisation: Organisation):
```
split up in 'with' and 'without' please, can be as simple as parametrizing the test.